package/kedro_viz/api/rest/router.py

```diff
@@ -49,6 +50,18 @@ async def get_single_node_metadata(node_id: str):
     return get_node_metadata_response(node_id)
 
 
+@router.post("/preferences")
+async def update_preferences(preferences: UserPreference):
+    try:
+        DataNodeMetadata.set_is_all_previews_enabled(preferences.showDatasetPreviews)
+        return {"message": "Preferences updated successfully"}
```

Comment:
You need to send the success message as - `return JSONResponse(status_code=200, content={"message": "Preferences updated successfully"})`. If not this will be printed as a message in the browser console.